src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,207 @@
+        for (int i = 0; i < table.length; i++){
+            if (table[i] instanceof Node) {
+                if(table[i].key instanceof Object && compareObjects(table[i].key, key)){
+                    table[i].value = value;
+                    return;
+                }
+                else if(table[i].key == key){
+                    table[i].value = value;
+                    return;
+                }
+            }
+        }
+
+        // If the key doesn't exist, we create a new Node
+        if (this.table[index] == null) {
+            Node node = new Node(h, key, value, null);
+            this.table[index] = node;
+        }
+        else{
+            Node oldNode = table[index];
+            if (oldNode.key!=key && oldNode.value!=value){
+                Node temp1 = oldNode;
+                Node temp2 = oldNode.next;
+                if (temp2 == null) temp1.next = new Node(h, key, value, null);
+                else{
+                    while (temp2!=null){
+                        temp1 = temp1.next;
+                        temp2 = temp2.next;
+                    }
+                    temp1.next = new Node(h, key, value, null);
+                }
+            }
+        }
+    }
     @Override
     public V getValue(K key) {
-        return null;
+
+        V result = null;
+
+        for (int i = 0; i < table.length; i++) {
+
+            Node node = table[i];
+
+            // If in this bucket there is only one Node we check it
+            if (node instanceof Node) {
+                if (node.next == null) {
+                    if (node.key instanceof Object && compareObjects(node.key, key)) {
+                        result = (V) node.value;
+                    } else if (node.key == key) {
+                        result = (V) node.value;
+                    }
+                }
+
+                // If in this bucket there is more than one Node we check all of them
+                else {
+                    while (node != null) {
+                        if (node.key instanceof Object && compareObjects(node.key, key)) {
+                            result = (V) node.value;
+                        } else if (node.key == key) {
+                            result = (V) node.value;
+                        }
+                        node = node.next;
+                    }
+                }
+
+            }
+
+        }
+        return result;
     }
 
     @Override
     public int getSize() {
-        return 0;
+
+        int size = 0;
+
+        for (int i = 0; i < table.length; i++){
+
+            Node node = table[i];
+
+            if (node instanceof Node) {
+                size++;
+            }
+
+            if (node instanceof Node && node.next != null) {
+
+                Node temp = node.next;
+
+                while (temp != null) {
+                    size++;
+                    temp = temp.next;
+                }
+            }
+
+        }
+        return size;
+    }
```

Comment:
Consider to store ```size``` as a property and increment it when new item is added to the map, so you will not need to calculate during every ```getSize``` method invocation